src/_pytest/python.py

```diff
@@ -1432,13 +1432,17 @@ def write_fixture(fixture_def: fixtures.FixtureDef[object]) -> None:
     def write_item(item: nodes.Item) -> None:
         # Not all items have _fixtureinfo attribute.
         info: Optional[FuncFixtureInfo] = getattr(item, "_fixtureinfo", None)
+        function: Optional[Callable[..., Any]] = getattr(item, "function", None)
         if info is None or not info.name2fixturedefs:
             # This test item does not use any fixtures.
             return
+        if function is None:
```

Comment:
Seems like the previous code assumed `item.function` exists. Maybe this should be `assert function is not None` then?

Reply:
Only python item have that